glacier_mapping/train.py

```diff
@@ -136,13 +145,30 @@ def log_images(writer, frame, batch, epoch, stage="train"):
     Return:
         Images Logged onto tensorboard
     """
-    pm = lambda x: x.permute(0, 3, 2, 1)
+    pm = lambda x: x.permute(0, 3, 1, 2)
     squash = lambda x: (x - x.min()) / (x.max() - x.min())
 
     x, y = batch
-    y_hat = torch.sigmoid(frame.infer(x))
+    if frame.multi_class:
+        y_hat = torch.nn.Softmax(3)(frame.infer(x))
+    else:
+        y_hat = torch.sigmoid(frame.infer(x))
+    y = torch.flatten(y.permute(0, 1, 3, 2), start_dim=2)
+    y_hat = torch.flatten(y_hat.permute(0, 1, 3, 2), start_dim=2)
     if epoch == 0:
-        writer.add_image(f"{stage}/x", make_grid(pm(squash(x))), epoch)
-        writer.add_image(f"{stage}/y", make_grid(pm(y)), epoch)
+        writer.add_image(f"{stage}/x", make_grid(pm(squash(x[:, :, :, :3]))), epoch)
+        writer.add_image(f"{stage}/y", make_grid(y.unsqueeze(1)), epoch)
 
-    writer.add_image(f"{stage}/y_hat", make_grid(pm(y_hat)), epoch)
+    writer.add_image(f"{stage}/y_hat", make_grid(y_hat.unsqueeze(1)), epoch)
+
+def update_metrics(main_metrics, batch_metrics):
+    for k, v in batch_metrics.items():
```

Comment:
You should probably just loop over values, right? The `k` here is immediately overwritten in the next line.

Reply:
not sure I understand. I take k (key) to update the base matrix. It doesn't get overwritten. It is concatenated at the location of k.

Reply:
Ah of course, I'm just going crazy, the next line is an `if` not a `for`.